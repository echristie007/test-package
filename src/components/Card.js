import PropTypes from 'prop-types';
import './Card.less';

/**
 * Card component presents a rectangle box layout with some data to fill it up with.
 */

const Card = ({ title, href, description, path, className }) => {
    function getFullPath() {
        let fullPath = '';
        path.forEach((pathSection, index) => {
            if (index === path.length - 1) {
                fullPath = fullPath.concat(String(pathSection));
            } else {
                fullPath = fullPath.concat(String(pathSection), '/');
            }
        });
        return fullPath;
    }

    return (
        <a className={`Card${className ? ' ' + className : ''}`} href={href}>
            <span className='Card__title'>{title}</span>
            {description ? <span className='Card__description'>{description}</span> : null}
            {path ? (
                <span>
                    <span className='Card__pathTitle'> Path </span>
                    <span className='Card__path'>{getFullPath()}</span>
                </span>
            ) : null}
        </a>
    );
};

Card.propTypes = {
    /** Title for the card */
    title: PropTypes.string.isRequired,

    /** Link is attached to the href attribute for the card */
    href: PropTypes.string.isRequired,

    /** Description inside in the card */
    description: PropTypes.string,

    /** Group to which the card belongs to */
    path: PropTypes.arrayOf(PropTypes.string),

    /** The component class name(s). */
    className: PropTypes.string,
};

export default Card;
