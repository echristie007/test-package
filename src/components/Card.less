@import '~@bluecateng/pelagos/less/spacing';
@import '~@bluecateng/pelagos/less/fonts';

.Card {
    display: flex;
    flex-direction: column;
    gap: @sp-08;
    background-color: var(--layer-01);
    width: 272px;
    height: 150px;
    padding: @sp-16;

    &__title {
        color: var(--link-primary);
        font-size: @sp-16;
        line-height: @sp-32;
        width: 184px;
        text-align: left;
    }

    &__description {
        display: -webkit-box;
        -webkit-line-clamp: 2;
        -webkit-box-orient: vertical;
        color: var(--text-placeholder);
        font-size: @sp-12;
        line-height: @sp-16;
        max-width: 235px;
        text-align: left;
        text-overflow: ellipsis;
        overflow: hidden;
    }

    &:hover {
        background-color: var(--layer-hover-01);
        outline: 1.5px solid var(--border-interactive);
        box-shadow: 0 0 5px @sp-02 var(--border-subtle-selected-01);
        text-decoration: none;
        .Card__title {
            color: var(--link-primary-hover);
            text-decoration: underline;
        }
    }
    &:active {
        background-color: var(--layer-active-01);
    }

    &__path {
        display: block;
        color: var(--text-primary);
        font-size: @sp-16;
        line-height: @sp-24;
        text-align: left;
    }

    &__pathTitle {
        display: block;
        color: var(--text-secondary);
        font-size: @sp-12;
        line-height: @sp-16;
        text-align: left;
    }
}
