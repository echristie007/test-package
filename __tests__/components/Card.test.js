import { shallow } from 'enzyme';
import Card from '../../src/components/Card';

jest.unmock('../../src/components/Card');

describe('Card', () => {
    describe('Rendering', () => {
        it('Render the Card component with default props', () => {
            const wrapper = shallow(<Card title='Add Host record' />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render the Card component with props', () => {
            const wrapper = shallow(
                <Card
                    href='#'
                    title='Add Host Record'
                    description='Add a host record on BAM'
                    path={['Host Record']}
                    className='test_cn'
                />,
            );
            expect(wrapper.getElement()).toMatchSnapshot();
        });
    });
});
