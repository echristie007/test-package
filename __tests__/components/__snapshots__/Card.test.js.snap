// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Rendering Render the Card component with default props 1`] = `
<a
  className="Card"
>
  <span
    className="Card__title"
  >
    Add Host record
  </span>
</a>
`;

exports[`Card Rendering Render the Card component with props 1`] = `
<a
  className="Card test_cn"
  href="#"
>
  <span
    className="Card__title"
  >
    Add Host Record
  </span>
  <span
    className="Card__description"
  >
    Add a host record on BAM
  </span>
  <span>
    <span
      className="Card__pathTitle"
    >
       Path 
    </span>
    <span
      className="Card__path"
    >
      Host Record
    </span>
  </span>
</a>
`;
